native/cocos/platform/android/java/src/com/cocos/lib/CanvasRenderingContext2DImpl.java

```diff
@@ -311,6 +319,25 @@ private void setLineJoin(String lineJoin) {
         mLineJoin = lineJoin;
     }
 
+    private void setShadowBlur(float blur) {
+        mShadowBlur = blur * 0.5f;
+    }
+
+    private void setShadowColor(float r, float g, float b, float a) {
+        mShadowColorR = (int) Math.ceil(r * 255.0f);
+        mShadowColorG = (int) Math.ceil(g * 255.0f);
+        mShadowColorB = (int) Math.ceil(b * 255.0f);
+        mShadowColorA = (int) Math.ceil(a * 255.0f);
```

Comment:
Divide with 255 in cpp, and multiply 255 in java. Why not just send original value to java?

Reply:
The value range of colors on apple is [0-1]

Reply:
So only divide on these platforms?

Reply:
Now all platforms perform divide

Reply:
Yep, that what i suggest that, only divide for needed platforms, not all platforms.